src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/ComponentParameterNullableWarningSuppressorTests.cs

```diff
@@ -0,0 +1,178 @@
+                {
+                    [Parameter]
+                    public string MyParameter { get; set; }
+                }
+                """;
+
+            await VerifyAnalyzerAsync(testCode,
+                // /0/Test0.cs(9,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(8, 19, 8, 30).WithSpan(8, 19, 8, 30).WithArguments("property", "MyParameter")
+                );
+        }
+
+        [Fact]
+        public async Task NoParameterRequiredStillReports()
+        {
+            var testCode = """
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+
+                public class MyComponent : ComponentBase
+                {
+                    [EditorRequired]
+                    public string MyParameter { get; set; }
+                }
+                """;
+
+            await VerifyAnalyzerAsync(testCode,
+                // /0/Test0.cs(9,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(8, 19, 8, 30).WithSpan(8, 19, 8, 30).WithArguments("property", "MyParameter")
+                );
+        }
+
+        [Fact]
+        public async Task AliasedAttributes()
+        {
+            var testCode = """
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+                using MyParameter = Microsoft.AspNetCore.Components.ParameterAttribute;
+                using MyRequired = Microsoft.AspNetCore.Components.EditorRequiredAttribute;
+
+                public class MyComponent : ComponentBase
+                {
+                    [MyParameter, MyRequired]
+                    public string MyParameter { get; set; }
+                }
+                """;
+
+            await VerifyAnalyzerAsync(testCode,
+                // /0/Test0.cs(9,19): warning CS8618: Non-nullable property 'MyParameter' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.
+                DiagnosticResult.CompilerWarning("CS8618").WithSpan(10, 19, 10, 30).WithSpan(10, 19, 10, 30).WithArguments("property", "MyParameter").WithIsSuppressed(true)
+                );
+        }
+
+        [Fact]
+        public async Task LocallyDefinedAttributes()
+        {
+            var testCode = """
+
+                #nullable enable
+                using System;
+                using Microsoft.AspNetCore.Components;
+                
+                public class MyComponent
```

Comment:
This test makes me wonder. Should we be checking that these attributes are being used in something that implements `IComponent`?

Reply:
I don't think that's required, consider this example:

```cs
abstract class B
{
    [Parameter, EditorRequired] string Param { get; set; }
}

class C : B, IComponent;
```

`B` doesn't implement `IComponent` but can still be instantiated as a component through `C`.

Reply:
IMO, for this type of edge case, I'd prefer to have false positives, rather than false negatives (iow, I think we should check for implementing `IComponent`). If we get feedback that this case is still being hit and causing pain, then we can take another look at it.

Reply:
Yeah, lets keep it tightly scoped for now, and we can expand it later if we get feedback on it